recipes/libsolace/all/conanfile.py

```diff
@@ -0,0 +1,65 @@
+import os
+
+from conans import CMake, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+from conans.model.version import Version
+
+class LibsolaceConan(ConanFile):
+    name = "libsolace"
+    description = "High performance components for mission critical applications"
+    topics = ("HPC", "High reliability", "P10", "solace", "performance", "c++", "conan")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/abbyssoul/libsolace"
+    license = "Apache-2.0"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False]
+    }
+    default_options = {"shared": False, "fPIC": True}
+    generators = "cmake"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _supported_cppstd(self):
+        return ["17", "gnu17", "20", "gnu20"]
+
+    def configure(self):
+        compiler_version = Version(str(self.settings.compiler.version))
+
+        if self.settings.os == "Windows":
+          raise ConanInvalidConfiguration("This library is not yet compatible with Windows")
+        if self.settings.compiler.cppstd and not self.settings.compiler.cppstd in self._supported_cppstd:
+          raise ConanInvalidConfiguration("This library requires c++17 standard or higher. {} required".format(self.settings.compiler.cppstd))
+        # Exclude GCC 6 that claims to support C++17 but does not in practice
+        if self.settings.compiler == "gcc" and (compiler_version < "7.0"):
```

Comment:
Nice, but about about clang and apple-clang?

Reply:
I believe clang and apple-clang are ok. At least versions that do support c++17, no?

Reply:
nope. Clang need to be 5.0 at least and apple-clang 9 IIRC:

```python
if (self.settings.compiler == "gcc" and compiler_version < "7") or \
   (self.settings.compiler == "clang" and compiler_version < "5") or \
   (self.settings.compiler == "apple-clang" and compiler_version < "9"):
    raise ConanInvalidConfiguration("This library requires c++17 standard or higher.") 
```

Reply:
believe me, Bincrafters was supporting clang 3.9 ... on my last job I was using gcc 4.1 for a project, so never doubt users' scenario haha!

Reply:
No worries. I was hoping that build system will fail the build of unsupported compilers :)
But yeah - my conan-build only runs for clang 7 and above: https://travis-ci.org/abbyssoul/conan-libsolace